src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -72,6 +75,65 @@ describeEachAppLayout(({ theme, size }) => {
     expect(wrapper.findDrawersTriggers()).toHaveLength(2);
   });
 
+  test.only('update rendered drawers via runtime config', async () => {
+    awsuiPlugins.appLayout.registerDrawer({ ...drawerDefaults, resizable: true });
+    const { wrapper } = await renderComponent(<AppLayout />);
+    // the 2nd trigger is for tools
+    expect(wrapper.findDrawersTriggers()).toHaveLength(2);
+    const drawerTrigger = wrapper.findDrawerTriggerById(drawerDefaults.id);
+
+    const triggerParent = drawerTrigger!.getElement().parentElement;
+    const triggerParentWrapper = createWrapper(triggerParent!);
+
+    awsuiPlugins.appLayout.updateDrawer({
+      id: drawerDefaults.id,
+      badge: true,
+    });
+    await delay();
+    const selector =
+      theme === 'refresh-toolbar'
+        ? toolbarTriggerStyles.badge
+        : theme === 'refresh' && size === 'desktop'
+          ? triggerStyles.badge
+          : iconStyles.badge;
+
+    expect(triggerParentWrapper?.findByClassName(selector)?.getElement()).toBeInTheDocument();
+
+    awsuiPlugins.appLayout.updateDrawer({
+      id: drawerDefaults.id,
+      badge: false,
+    });
+    await delay();
+
+    // re-querying the dot element
+    expect(triggerParentWrapper?.findByClassName(selector)?.getElement()).toBeUndefined();
```

Comment:
This line needs some strictness

```suggestion
    expect(triggerParentWrapper!.findByClassName(selector)).toBeUndefined();
```

1. `triggerParentWrapper` is never undefined
2. If `findByClassName(selector)` returns no match, that `getElement()` is never called anyway.
For symmetry reasons, I suggest to replace `?` with `!` in the other matchers too

Reply:
updated.